sopel/tools/time.py

```diff
@@ -40,25 +75,32 @@ def validate_timezone(zone):
     becomes ``UTC``. In the majority of user-facing interactions, such
     case-insensitivity will be expected.
 
-    If the zone is not valid, ``ValueError`` will be raised.
+    If the zone is not valid, :exc:`ValueError` will be raised.
 
     .. versionadded:: 6.0
+
+    .. versionchanged:: 8.0
+
+        If ``zone`` is ``None``, raises a :exc:`ValueError` as if it was an
+        empty string or an invalid timezone instead of returning ``None``.
+
     """
     if zone is None:
-        return None
+        raise ValueError('Invalid time zone.')
```

Comment:
The idea is not to add another exception type that must be caught. Strictly speaking, yes, `NoneType != str` and this case should be a `TypeError`, but existing code that uses this function expects anything invalid to raise `ValueError`.

Then again, we're _already_ changing the semantics by raising an exception on `None` instead of just returning `None`, so maybe there's actually no valid argument here for "keeping compatibility with old code" after all.

Reply:
In theory, yes. However, this is legacy code, so changes are limited to what fit best the existing behavior.